File: ee25btech11011/MATGEO/1_8_28/codes/code.py

import math

# Define points A and B
A = (3, 4, -5)
B = (-2, 1, 4)

# Midpoint M = (A+B)/2
M = tuple((a + b) / 2 for a, b in zip(A, B))

# Vector AB = B - A
AB = tuple(b - a for a, b in zip(B, A))

# |AB|^2
AB_sq = sum(x**2 for x in AB)

# |A-M|^2 = |AB|^2 / 4
AM_sq = AB_sq / 4

# Input k
k = float(input("Enter value of k: "))

# Radius^2
radius_sq = k*k - AM_sq

print("\n--- Solution for locus of P ---")
print(f"A = {A},  B = {B}")
print(f"Midpoint M = {M}")
print(f"|AB|^2 = {AB_sq}")
print(f"|A-M|^2 = {AM_sq:.2f}")

if radius_sq < 0:
    print("\nNo real sphere exists because radius^2 < 0.")
else:
    print("\nEquation of sphere:")
    print(f"(x - {M[0]})^2 + (y - {M[1]})^2 + (z - {M[2]})^2 = {radius_sq}")
    print(f"Centre: {M}")
    print(f"Radius: {math.sqrt(radius_sq):.4f}")

File: ee25btech11011/MATGEO/1_8_28/codes/code.c
#include <stdio.h>
#include <math.h>

int main() {
    // Define points A and B
    double A[3] = {3, 4, -5};
    double B[3] = {-2, 1, 4};
    double M[3];   // midpoint

    // Midpoint M = (A+B)/2
    M[0] = (A[0] + B[0]) / 2.0;
    M[1] = (A[1] + B[1]) / 2.0;
    M[2] = (A[2] + B[2]) / 2.0;

    // Vector AB = B - A
    double AB[3];
    AB[0] = B[0] - A[0];
    AB[1] = B[1] - A[1];
    AB[2] = B[2] - A[2];

    // |AB|^2
    double AB_sq = AB[0]*AB[0] + AB[1]*AB[1] + AB[2]*AB[2];

    // |A-M|^2 = |AB|^2 / 4
    double AM_sq = AB_sq / 4.0;

    // Input k
    double k;
    printf("Enter value of k: ");
    scanf("%lf", &k);

    // Radius^2
    double radius_sq = k*k - AM_sq;

    printf("\n--- Solution for locus of P ---\n");
    printf("Midpoint M = (%.2f, %.2f, %.2f)\n", M[0], M[1], M[2]);
    printf("|AB|^2 = %.2f\n", AB_sq);
    printf("|A-M|^2 = %.2f\n", AM_sq);

    if (radius_sq < 0) {
        printf("\nNo real sphere exists because radius^2 < 0.\n");
    } else {
        printf("\nEquation of sphere:\n");
        printf("(x - %.2f)^2 + (y - %.2f)^2 + (z - %.2f)^2 = %.2f\n",
               M[0], M[1], M[2], radius_sq);
        printf("Centre: (%.2f, %.2f, %.2f)\n", M[0], M[1], M[2]);
        printf("Radius: %.4f\n", sqrt(radius_sq));
    }

    return 0;
}